api/src/main/kotlin/edu/wgu/osmt/collection/CollectionController.kt

```diff
@@ -50,196 +55,329 @@ import org.springframework.web.util.UriComponentsBuilder
-            ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
-
+                ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+        
         val updated = collectionRepository.updateFromApi(
-            existing.id.value,
-            apiUpdate,
-            richSkillRepository, oAuthHelper.readableUserName(user)
+                existing.id.value,
+                apiUpdate,
+                richSkillRepository, oAuthHelper.readableUserName(user)
         )
-            ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
-
+                ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+        
         return ApiCollection.fromDao(updated, appConfig)
     }
-
-    @PostMapping(RoutePaths.COLLECTION_SKILLS_UPDATE, produces = [MediaType.APPLICATION_JSON_VALUE])
+    
+    @PostMapping(path = [
+        "${RoutePaths.API}${RoutePaths.API_V2}${RoutePaths.COLLECTION_UPDATE}",
+        "${RoutePaths.API}${RoutePaths.UNVERSIONED}${RoutePaths.COLLECTION_UPDATE}",
+        ]
+        , produces = [MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun updateCollectionV2(
+            @PathVariable uuid: String,
+            @RequestBody apiUpdate: ApiCollectionUpdate,
+            @AuthenticationPrincipal user: Jwt?
+    ): ApiCollection {
+        
+        return ApiCollectionV2.fromLatest(updateCollection(uuid, apiUpdate, user), appConfig)
+    }
+    
+    @PostMapping(path = [
+        "${RoutePaths.API}/{apiVersion}${RoutePaths.COLLECTION_SKILLS_UPDATE}"
+                        ],
+            produces = [MediaType.APPLICATION_JSON_VALUE])
     @ResponseBody
     fun updateSkills(
-        @PathVariable uuid: String,
-        @RequestBody skillListUpdate: ApiSkillListUpdate,
-        @RequestParam(
-            required = false,
-            defaultValue = PublishStatus.DEFAULT_API_PUBLISH_STATUS_SET
-        ) status: List<String>,
-        @AuthenticationPrincipal user: Jwt?
+            @PathVariable(name = "apiVersion", required = false) apiVersion: String?,
+            @PathVariable uuid: String,
+            @RequestBody skillListUpdate: ApiSkillListUpdate,
+            @RequestParam(
+                    required = false,
+                    defaultValue = PublishStatus.DEFAULT_API_PUBLISH_STATUS_SET
+            ) status: List<String>,
+            @AuthenticationPrincipal user: Jwt?
     ): HttpEntity<TaskResult> {
         val publishStatuses = status.mapNotNull { PublishStatus.forApiValue(it) }.toSet()
-        val task = UpdateCollectionSkillsTask(uuid, skillListUpdate, publishStatuses=publishStatuses, userString = oAuthHelper.readableUserName(user))
+        val task = if(RoutePaths.API_V3 == RoutePaths.getApiVersionCalled(apiVersion)) {
+            UpdateCollectionSkillsTask(uuid, skillListUpdate, publishStatuses = publishStatuses, userString = oAuthHelper.readableUserName(user), apiResultPath = "${RoutePaths.API}${RoutePaths.API_V3}${RoutePaths.TASK_DETAIL_BATCH}")
 
+        } else {
```

Comment:
This should be:
`else if (!apiVersion || RoutePaths.API_V2 == "${apiVersion.lowercase()}/")`

And then we need a separate else block to catch v4, v5, etc.